README.md

```diff
@@ -17,42 +17,137 @@ packages, operating systems, etc.
+* `node['osl-git']['secrets]'['credentials']`
+
+The logic for choosing databags vs attributes is in `libraries/default.rb`
+
+The full syntax for all of the resource's properties is:
+```ruby
+git_credentials 'name' do
+  file                 String # default value: /root/.git-credentials
+  secrets_databag      String # default value: node['osl-git']['secrets_databag']
+  secrets_item         String # default value: node['osl-git']['secrets_item']
+  action               Symbol # defaults to :create
+end
+```
+
+An example where git credentials are stored in an alternative path and a non-default databag
+item is used:
+```ruby
+git_credentials 'default' do
+  file '/root/.config/git/credentials'
+  secrets_databag 'secrets'
+  secrets_item 'git_tokens'
+end
+```
 
-e.g.
-Just include `osl-git` in your node's `run_list`:
+#### Actions
+`:create`
+
+Enable the git credentials store and populates `file` with credentials from
+`secrets_databag:secrets_item` (or from `node['osl-git']['secrets']['credentials']` in a
+testing environment).
+
+`:trigger`
+
+Mark the resource as having been updated to trigger notifications. (Used in the resource code
+to work around [chef#6987](https://github.com/chef/chef/issues/6987).)
+
+Enable the git credentials store and populates `file` with credentials from
+`secrets_databag:secrets_item` (or from `node['osl-git']['secrets']['credentials']` in a
+testing environment).
+
+`:nothing`
+
+Do nothing until notified by another resource.
+
+#### Properties
+`file` - **Ruby Type:** String | **Default Value:** `/root/.git-credentials`
+
+The path to store credentials in that the git credentials store will read from.
+
+`secrets_databag` - **Ruby Type:** String | **Default Value:** `node['osl-git']['secrets_databag']`
+
+The name of the databag where credentials are stored.
+
+`secrets_item` - **Ruby Type:** String | **Default Value:** `node['osl-git']['secrets_item']`
+
+The name of the databag item where credentials are stored.
+
+Databag
+-------
+The `git_credentials` resource expects a `credentials` list in databag items:
 
 ```json
 {
-  "name":"my_node",
-  "run_list": [
-    "recipe[osl-git]"
+  "id": "git_tokens",
+  "credentials": [
+    "https://username:token@github.com",
+    "foo:bar@gitlab.com"
   ]
 }
 ```
 
+Testing
+-------
+To test that stored credentials can actually be used to clone a private repostiory, you must set
+up your own credentials for the `git-credentials-attributes` and `git-credentials-databag` test
+suites.
+
+**Note:** If you are not a part of the Open Source Lab, you will have to change the repo in the
+test cookbook to a private repo that you have access to.
+
+For `git-credentials-attributes` add your credentials to the list in `.kitchen.yml` or set your
+GitHub token in the `GITHUB_TOKEN` environmental variable.
+
+For `git-credentials-databag` add your credentials to one of the databags in
+`test/integration/data_bags/osl-git`.
+
```

Comment:
I think it might be a better idea to create a test repo and a deploy key that only has read-only access. That way we don't have to deal with changing things. We can just put it on git.o.o. Thoughts?

Reply:
Sounds good to me.